x-pack/plugins/lens/public/datatable_visualization/components/table_basic.tsx

```diff
@@ -0,0 +1,245 @@
+    updateTable(firstTable);
+  }, [firstTable]);
+
+  const firstTableRef = useRef(firstLocalTable);
+  firstTableRef.current = firstLocalTable;
+
+  const hasAtLeastOneRowClickAction = props.rowHasRowClickTriggerActions?.some((x) => x);
+
+  const { getType, dispatchEvent, renderMode, formatFactory } = props;
+
+  const formatters: Record<string, ReturnType<FormatFactory>> = useMemo(
+    () =>
+      firstLocalTable.columns.reduce(
+        (map, column) => ({
+          ...map,
+          [column.id]: formatFactory(column.meta?.params),
+        }),
+        {}
+      ),
+    [firstLocalTable, formatFactory]
+  );
+
+  const onClickValue = useCallback(
+    (data: LensFilterEvent['data']) => {
+      dispatchEvent({ name: 'filter', data });
+    },
+    [dispatchEvent]
+  );
+
+  const onEditAction = useCallback(
+    (data: LensSortAction['data'] | LensResizeAction['data']) => {
+      if (renderMode === 'edit') {
+        dispatchEvent({ name: 'edit', data });
+      }
+    },
+    [dispatchEvent, renderMode]
+  );
+  const onRowContextMenuClick = useCallback(
+    (data: LensTableRowContextMenuEvent['data']) => {
+      dispatchEvent({ name: 'tableRowContextMenuClick', data });
+    },
+    [dispatchEvent]
+  );
+
+  const handleFilterClick = useMemo(() => createGridFilterHandler(firstTableRef, onClickValue), [
+    firstTableRef,
+    onClickValue,
+  ]);
+
+  const bucketColumns = useMemo(
+    () =>
+      columnConfig.columnIds.filter((_colId, index) => {
+        const col = firstTableRef.current.columns[index];
+        return (
+          col?.meta?.sourceParams?.type &&
+          getType(col.meta.sourceParams.type as string)?.type === 'buckets'
+        );
+      }),
+    [firstTableRef, columnConfig, getType]
+  );
+
+  const isEmpty =
+    firstLocalTable.rows.length === 0 ||
+    (bucketColumns.length &&
+      firstTable.rows.every((row) => bucketColumns.every((col) => row[col] == null)));
+
+  const visibleColumns = useMemo(() => columnConfig.columnIds.filter((field) => !!field), [
+    columnConfig,
+  ]);
+
+  const { sortBy, sortDirection } = columnConfig;
+
+  const isReadOnlySorted = renderMode !== 'edit';
+
+  const onColumnResize = useMemo(
+    () => createGridResizeHandler(columnConfig, setColumnConfig, onEditAction),
+    [onEditAction, setColumnConfig, columnConfig]
+  );
+
+  const columns: EuiDataGridColumn[] = useMemo(
+    () =>
+      createGridColumns(
+        bucketColumns,
+        firstLocalTable,
+        handleFilterClick,
+        isReadOnlySorted,
+        columnConfig,
+        visibleColumns,
+        formatFactory,
+        onColumnResize
+      ),
+    [
+      bucketColumns,
+      firstLocalTable,
+      handleFilterClick,
+      isReadOnlySorted,
+      columnConfig,
+      visibleColumns,
+      formatFactory,
+      onColumnResize,
+    ]
+  );
+
+  const trailingControlColumns: EuiDataGridControlColumn[] = useMemo(() => {
+    if (!hasAtLeastOneRowClickAction || !onRowContextMenuClick) {
```

Comment:
This section disappears when in edit mode. Is that expected?

Reply:
It was working like this before too so I think it's fine...